### PR TITLE
fix 11057 by eliminating shortName class check and add log

### PR DESCRIPTION
**Pull request by niravassar**
This is a branch of 3.3.x for #11057

### CODE REVIEW

File: grails-core/src/main/groovy/org/grails/plugins/AbstractGrailsPluginManager.java [hunk 1]
```diff
@@ -315,7 +323,7 @@ public void registerProvidedArtefacts(GrailsApplication app) {
     }
 
     private boolean isAlreadyRegistered(GrailsApplication app, Class<?> artefact, String shortName) {
```
Comment (jameskleeh) [inline grails-core/src/main/groovy/org/grails/plugins/AbstractGrailsPluginManager.java]: The `shortName` can be removed from this method now

File: grails-core/src/main/groovy/org/grails/plugins/AbstractGrailsPluginManager.java [hunk 2]
```diff
@@ -306,6 +306,14 @@ public void registerProvidedArtefacts(GrailsApplication app) {
                 if(isPluginDisabledForProfile(plugin)) continue;
                 for (Class<?> artefact : plugin.getProvidedArtefacts()) {
                     String shortName = GrailsNameUtils.getShortName(artefact);
+                    if (artefact.getName().equals(shortName)) {
+                        LOG.error("Plugin " + plugin.getName() + " has an artefact " + shortName + " without a package name " +
```
Comment (jameskleeh) [inline grails-core/src/main/groovy/org/grails/plugins/AbstractGrailsPluginManager.java]: This should be a warning

File: grails-core/src/main/groovy/org/grails/plugins/AbstractGrailsPluginManager.java (hunk 1 above)
Comment (niravassar) [reply grails-core/src/main/groovy/org/grails/plugins/AbstractGrailsPluginManager.java]: done

File: grails-core/src/main/groovy/org/grails/plugins/AbstractGrailsPluginManager.java (hunk 2 above)
Comment (niravassar) [reply grails-core/src/main/groovy/org/grails/plugins/AbstractGrailsPluginManager.java]: done

**Pull request merged**